### PR TITLE
Add support for right aligned sub navigation in `dotcom-ui-header` 

### PR DESCRIPTION
**Pull request by notlee**
Adds a `subsections-right` array to the `THeaderProps` along with the
existing `subsections` array, with items which are positioned to the
right of the subnav. This will be used my `next-profile` initially
to align a signout link to the right of the subnav. There is no
requirement for this to be defined in navigation apis (Origami
Navigation Service or Next Navigation API) but could be added
in the future without a breaking change to the APIs.

Related o-header PR: Financial-Times/o-header#395
Related issue: Financial-Times/o-header#376

Screenshot 2020-09-03 at 13 56 02

### CODE REVIEW

**Review by alexnaish — APPROVED** [review]
LGTM 🚀

File: packages/dotcom-ui-header/src/__stories__/story.tsx [hunk 1]
```diff
@@ -29,7 +37,6 @@ storiesOf('FT / Header', module)
   .add('Default header with drawer', () => {
     const knobs = {
       showSubNavigation: toggleShowSubNav(),
-      showUserNavigation: toggleUserStateOptions(),
       showMegaNav: toggleShowMegaNav(),
```
Comment (notlee) [inline packages/dotcom-ui-header/src/__stories__/story.tsx:40]: Whoops: Restore the user navigation knob for the "default header with drawer"  `dotcom-ui-header` storybook story
```suggestion
      showMegaNav: toggleShowMegaNav(),
      showUserNavigation: toggleUserStateOptions(),
```

**Review by andygout — APPROVED** [review]
Code looks good and I have run Storybook locally to check working as described there too and it's all 👍.

There might be conflicts with https://github.com/Financial-Times/dotcom-page-kit/pull/853, so I think #853 should be merged first as it is the smaller of the two and the changes are extremely clear-cut.

**Comment by andygout** [comment]
Turned out to be a seamless rebase 👍.

**Comment by notlee** [comment]
thanks @andygout!

**Pull request merged**